### PR TITLE
feat: add lmdb database in Storage

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

File: src/block.zig
Comment (AbdelStark) [inline src/block.zig:1]: Please move to `src/types/block.zig`

File: src/block.zig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const std = @import("std");
+
+pub const Block = struct {
```
Comment (AbdelStark) [inline src/block.zig:3]: add Zig doc on struct and fields

File: src/storage/storage.zig [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
-    /// - `self`: Storage
-    pub fn deinit(self: *Storage) void {
-        // Clean up resources if needed
-        _ = self;
+    /// Release the lmdb environment handle.
+    pub fn deinit(self: Storage) void {
+        self.env.deinit();
+    }
+
+    pub fn init_transaction(self: Storage) !Transaction {
```
Comment (AbdelStark) [inline src/storage/storage.zig:38]: add doc

File: src/storage/storage.zig [hunk 3]
```diff
@@ -1,34 +1,55 @@
...
+        const txn = try lmdb.Transaction.init(self.env, .{ .mode = .ReadWrite });
+        return Transaction{ .txn = txn };
+    }
+};
+
+pub const Transaction = struct {
```
Comment (AbdelStark) [inline src/storage/storage.zig:44]: add doc for struct and functions

**Review by AbdelStark — CHANGES_REQUESTED** [review]
left minor comments

File: src/block.zig
Comment (tdelabro) [reply src/block.zig:1]: done

File: src/block.zig (hunk 1 above)
Comment (tdelabro) [reply src/block.zig]: done

File: src/storage/storage.zig (hunk 2 above)
Comment (tdelabro) [reply src/storage/storage.zig:39]: done

File: src/storage/storage.zig (hunk 3 above)
Comment (tdelabro) [reply src/storage/storage.zig:46]: done

**Pull request merged**